### DISCUSSION TITLE
How to create custom block

### DISCUSSION THREAD

**Discussion by X3Technology** [vanjarosoftware/Vanjaro.Platform#1465] [Q&A]
I am playing around trying to learn how to create my own custom blocks.

For a test, I was trying to create a new Block that would create a BootStrap5 Navbar
https://getbootstrap.com/docs/5.0/components/navbar/

This is essentially just a combination of the current Logo block and the Menu block, so I figured all of the pieces should already be there.  However, the HTML markup is just a bit different as the `<nav>` tag wraps the entire section (including the logo) instead of just the menu component.

To get started, I tried just copy/pasting the Menu folder found in '/Portals/vThemes/Basic/blocks' and renaming the folder to 'Navbar', but my new block doesn't show up in the UX yet so there must be something more I need to do.  

Any guidance would be appreciated.

**Comment by meetmandeep** [comment]
Design Blocks are server side blocks and use Razor Pages. You need to implement an interface. I would recommend holding off until we have  some documentation in place. Our focus right now is to get the official release out and then we'll start creating tutorials, videos, and webinars on development.
